addons/binding/org.openhab.binding.nibeuplink/src/main/java/org/openhab/binding/nibeuplink/handler/GenericUplinkHandler.java

```diff
@@ -0,0 +1,219 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.nibeuplink.config.NibeUplinkConfiguration;
+import org.openhab.binding.nibeuplink.internal.connector.UplinkWebInterface;
+import org.openhab.binding.nibeuplink.internal.model.Channel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link GenericUplinkHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+public abstract class GenericUplinkHandler extends BaseThingHandler implements NibeUplinkHandler {
+
+    private final String NO_VALUE = "--";
+
+    private final Logger logger = LoggerFactory.getLogger(GenericUplinkHandler.class);
+
+    private Set<Channel> deadChannels = new HashSet<>(1000);
+
+    /**
+     * Refresh interval which is used to poll values from the NibeUplink web interface (optional, defaults to 60 s)
+     */
+    private int refreshInterval;
+
+    /**
+     * Refresh interval which is used clean the dead channel list (optional, defaults to 1 h)
+     */
+    private int houseKeepingInterval = 1;
+
+    /**
+     * Interface object for querying the FRITZ!Box web interface
+     */
+    private UplinkWebInterface webInterface;
+
+    /**
+     * Job which will do the FRITZ!Box polling
+     */
+    private final UplinkPolling pollingRunnable;
+
+    /**
+     * Schedule for polling
+     */
+    private ScheduledFuture<?> pollingJob;
+
+    /**
+     * Schedule for periodic cleaning dead channel list
+     */
+    private ScheduledFuture<?> deadChannelHouseKeeping;
+
+    public GenericUplinkHandler(@NonNull Thing thing) {
+        super(thing);
+        this.pollingRunnable = new UplinkPolling(this);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("command for {}: {}", channelUID, command);
+
+        // TODO: necessary??? will be read only in the first place
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("About to initialize NibeUplink");
+        Thing thing = this.getThing();
+        NibeUplinkConfiguration config = getConfiguration();
+
+        logger.debug("Discovered NibeUplink initialized: {}", config);
+
+        this.refreshInterval = config.getPollingInterval();
+        this.houseKeepingInterval = config.getHouseKeepingInterval();
+        this.webInterface = new UplinkWebInterface(config, this);
+
+        if (config.getPassword() != null) {
```

Comment:
In `UplinkWebInterface` a  preCheck already checks this, but also some other values. If other values would lead to configuration value (or password is empty) then it will still start polling.
Also because password is required it will probably never reach makes this false because openHAB will not call this initialize method if the password was not set.

Reply:
removed this check